doc/sacred.md

```diff
@@ -0,0 +1,175 @@
+
+ - Using **[dicts](https://sacred.readthedocs.io/en/stable/configuration.html#dictionaries)**: Set the configuration
+  directly as a dictionary.
+ - Using **[Configuration files](https://sacred.readthedocs.io/en/stable/configuration.html#config-files)**: Sacred
+  supports loading configuration from JSON and yaml files.
+ - Using **[ConfigScopes](https://sacred.readthedocs.io/en/stable/configuration.html#config-scopes)**: ConfigScopes are 
+  functions wrapped by sacred with the `config` decorator. All local
+  variables in these functions are interpreted as configuration values.
+  This enables complex computations, dependencies and automatic filling of missing configuration values.
+  This is the recommended way of setting configuration values in the code.
+ - Using **[named ConfigScopes](https://sacred.readthedocs.io/en/stable/configuration.html#named-configurations)**: 
+  You can bundle frequently used non-default configuration in config scopes and give
+  them a name with the `named_config` decorator. This way you can call your script `with named_config_name` and
+  sacred inserts the config.
+ - Via the **[command line interface](https://sacred.readthedocs.io/en/stable/command_line.html#configuration-updates)**:
+  You can call your script with the arguments `with configuration_key=value` and
+  it automatically updates all configuration values that depend on `configuration_key` and stores information about
+  what was modified.
+For more information about how sacred handles configuration, have a look at the documentation about 
+[the config](https://sacred.readthedocs.io/en/stable/configuration.html) and 
+[the command line interface](https://sacred.readthedocs.io/en/stable/command_line.html).
+ 
+Consider the following example for a trainings script using sacred:
+
+```python
+from sacred import Experiment
+from sacred.observers import FileStorageObserver
+import padertorch as pt
+from pathlib import Path
+
+# Define the experiment. Experiment is the main class for managing experiments.
+ex = Experiment('my-experiment-name')
+
+
+class MyModel(pt.Model):
+    # Implement your model here
+    def __init__(self, num_layers=2):
+        super().__init__()
+        self.num_layers = num_layers
+
+
+# Create a ConfigScope with the config decorator
+@ex.config
+def config():
+    """
+    This function contains the configuration of the experiment. All local 
+    variables in this function will be captured by sacred and used as 
+    configuration values.
+    """
+    # padertorch.Trainer inherits from Configurable, so we can easily construct 
+    # the config dictionary here and instantiate the trainer from the config 
+    # later
+    trainer = {
+        # The 'factory' key tells the Configurable which class to instantiate
+        'factory': pt.Trainer,
+        # The next line creates a new experiment folder for this experiment
+        'storage_dir': pt.io.get_new_storage_dir(ex.path),
```

Comment:
The default value of the `mkdir` argument of `pt.io.get_new_storage_dir` is True.
For a resume or cli overwrite it is better, when you do something like:
```python
    trainer = {..., 'storage_dir': None, ...}
    if trainer['storage_dir'] is None:
        trainer['storage_dir'] = pt.io.get_new_storage_dir(ex.path)
```
It looks not so nice, but works better.